### PR TITLE
Activate Py 3.10 builds

### PR DESCRIPTION
**Pull request by joshmoore**
With the freeze for Python 3.11 (https://www.python.org/downloads/release/python-3110b1/), I thought it would make sense to at least try out the latest in a PR. We likely will want to re-evaluate the matrix and perhaps set some items to optional.

### CODE REVIEW

**Comment by jakirkham** [comment]
Guessing there are missing dependencies

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -15,11 +15,15 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11.0-alpha - 3.11.0']
         numpy_version: ['!=1.21.0', '==1.17.*']
         exclude:
           - python-version: 3.9
             numpy_version: '==1.17.*'
+          - python-version: 3.10
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply .github/workflows/python-package.yml:23]: Almost certainly, thanks.

**Comment by joshmoore** [comment]
🤦🏽 Was reading setup-python documentation but we use setup-conda. I'll keep digging. Worst case, I'll create a simple workflow that uses setup-python.

**Comment by joshmoore** [comment]
Pushed directly to `origin` to test (see https://github.com/zarr-developers/zarr-python/runs/6385329519?check_suite_focus=true) but of course now we're getting into building wheels for numpy, numcodecs, etc.

Likely just too premature, sadly.

**Comment by jakirkham** [comment]
Yeah unfortunately. There are people starting to look into it

That said, adding Python 3.10 makes a lot of sense

**Comment by jakirkham** [comment]
Thanks Josh! 🙏

**Comment by jakirkham** [comment]
Also just as a note, we added Python 3.10 to Numcodecs in PR ( https://github.com/zarr-developers/numcodecs/pull/318 )

**Pull request merged**